pkg/tnf/handlers/deploymentsdrain/deploymentsdrain.go

```diff
@@ -0,0 +1,95 @@
+// Copyright (C) 2021 Red Hat, Inc.
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+package deploymentsdrain
+
+import (
+	"time"
+
+	"github.com/test-network-function/test-network-function/pkg/tnf"
+	"github.com/test-network-function/test-network-function/pkg/tnf/identifier"
+	"github.com/test-network-function/test-network-function/pkg/tnf/reel"
+)
+
+const (
+	ddRegex                = "SUCCESS"
+	drainTimeoutPercentage = 90 // drain timeout is a percentage of test timeout
+)
+
+// DeploymentsDrain holds information derived from running "oc adm drain" on the command line.
+type DeploymentsDrain struct {
+	result  int
+	timeout time.Duration
+	args    []string
+}
+
+// NewDeploymentsDrain creates a new DeploymentsDrain tnf.Test.
+func NewDeploymentsDrain(timeout time.Duration, node string) *DeploymentsDrain {
+	drainTimeout := timeout * drainTimeoutPercentage / 100
```

Comment:
Just in case others are reviewing and wondering;  it would appear that time.Duration is smart enough to handle `float-like` numbers and possible type-upgrades even when multiplying by `int` types.  This is the type of Go operator overloading behavior I am not super familiar with, and had to punch it into the playground to confirm:

https://play.golang.org/p/2bMweG0-5jZ

Reply:
I am not sure what you're trying to focus on here.
If you are referring to the Duration type's ability to hold float then the answer is yes.
I read docs and tested.

Reply:
Yes, just leaving it in case Charlie or anyone else reviewing was curious.  No action is needed.